server/routes/basicDetails.ts

```diff
@@ -0,0 +1,411 @@
+      breachNotice.useDefaultAddress = null
+    }
+    // reply address
+    if (req.body.replyAddressSelectOne === 'No') {
+      breachNotice.replyAddress = getSelectedAddress(basicDetails.replyAddresses, req.body.alternateReplyAddress)
+      breachNotice.useDefaultReplyAddress = false
+    } else if (req.body.replyAddressSelectOne === 'Yes') {
+      breachNotice.replyAddress = defaultReplyAddress
+      breachNotice.useDefaultReplyAddress = true
+    } else {
+      breachNotice.useDefaultReplyAddress = null
+    }
+
+    const combinedName: string = `${basicDetails.title} ${basicDetails.name.forename} ${basicDetails.name.middleName} ${basicDetails.name.surname}`
+    breachNotice.titleAndFullName = combinedName.trim().replace('  ', ' ')
+
+    // validation
+    const errorMessages: ErrorMessages = validateBasicDetails(breachNotice, req.body.dateOfLetter)
+    const hasErrors: boolean = Object.keys(errorMessages).length > 0
+
+    if (hasErrors) {
+      const alternateAddressOptions = initiateAlternateAddressSelectItemList(basicDetails, breachNotice)
+      const replyAddressOptions = initiateReplyAddressSelectItemList(basicDetails, breachNotice)
+      const basicDetailsDateOfLetter: string = req.body.dateOfLetter
+      res.render(`pages/basic-details`, {
+        errorMessages,
+        breachNotice,
+        basicDetails,
+        defaultOffenderAddress,
+        defaultReplyAddress,
+        alternateAddressOptions,
+        replyAddressOptions,
+        basicDetailsDateOfLetter,
+      })
+    } else {
+      // mark that a USER has saved the document at least once
+      breachNotice.basicDetailsSaved = true
+      await breachNoticeApiClient.updateBreachNotice(breachNoticeId, breachNotice)
+      console.log(req.body)
+      console.log(`action is ${req.body.action}`)
+
+      if (req.body.action === 'viewDraft') {
+        try {
+          await showDraftPdf(breachNotice.id, res)
+        } catch (err) {
+          // Render the page with the new error
+          errorMessages.pdfRenderError = {
+            text: 'There was an issue generating the draft report. Please try again or contact support.',
+          }
+
+          const alternateAddressOptions = initiateAlternateAddressSelectItemList(basicDetails, breachNotice)
+          const replyAddressOptions = initiateReplyAddressSelectItemList(basicDetails, breachNotice)
+          const basicDetailsDateOfLetter: string = req.body.dateOfLetter
+          res.render(`pages/basic-details`, {
+            errorMessages,
+            breachNotice,
+            basicDetails,
+            defaultOffenderAddress,
+            defaultReplyAddress,
+            alternateAddressOptions,
+            replyAddressOptions,
+            basicDetailsDateOfLetter,
+          })
+        }
+      } else if (req.body.action === 'saveProgressAndClose') {
+        res.setHeader('Content-Security-Policy', "script-src-elem 'unsafe-inline'")
+        res.send('<script>window.close();</script>')
```

Comment:
Bit dodgy, but if it works 😄 

Might be better to add the CSP nonce to the script tag to mark it as trusted, which would save you overriding the CSP header:
```suggestion
        res.send(`<script nonce="${res.locals.cspNonce}">window.close()</script>`)
```